### DISCUSSION TITLE
Contentful Starter Query Error

### DISCUSSION THREAD

**Discussion by adanaher** [vercel/next.js#33642] [Help]
I'm trying to set up the Contentful Next.js starter but am getting the following error for the GraphQL query inside the getAllPostsForHome function:

'Query cannot be executed. The maximum allowed complexity for a query is 11000 but it was 100200. Simplify the query e.g. by setting lower limits for collections.'

I got the same error when I tried to run the query in Postman. I should note that I only have the mock entries in Contentful - 1 author and 2 posts with images and some dummy plaintext as content, so it's not an issue of having too much content. I also removed the author field from the query because I didn't need that, so now it seems to be the nested sub-fields under the content field that are causing the problem since when I remove some of them from the query it works fine.

**Comment by adanaher** [comment]
I think I solved this one for myself. The issue seems to be that if you don't set limits on the rich text field then the complexity can easily exceed the API limits because the complexity is based on the number of linked assets allowed. See https://www.contentful.com/developers/docs/references/graphql/#/introduction/query-complexity-limits.

So, there seems to be 2 ways to fix this. The quick one (if you're just trying to get the sample up and running) is to add some limits into the GraphQL queries in the lib/api.js file in the starter. The better ways seems to be to update the settings for the content model of the Content field of the Post type to limit the number of the linked entities, blocks, etc. In the validation tab of the settings, I just checked all of the "Limit number of..." checkboxes and set the upper limit to 5. I'll adjust that as I create blog posts if I need to.

**Reply by marcelkalveram** [reply]
Same here. I solved this by adding a limit of `10` to the `getAllPosts` function:

```
export async function getAllPosts(isDraftMode: boolean): Promise<any[]> {
  const entries = await fetchGraphQL(
    `query {
      postCollection(where: { slug_exists: true }, order: date_DESC, limit: 10, preview: ${
        isDraftMode ? "true" : "false"
      }) {
        items {
          ${POST_GRAPHQL_FIELDS}
        }
      }
    }`,
    isDraftMode
  );
  return extractPostEntries(entries);
}
```
